jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ResponseWriter.java

```diff
@@ -171,7 +173,10 @@ public void complete(Callback callback)
         {
             _isClosed = true;
         }
-        _httpWriter.complete(callback);
+        if (_writer instanceof AbstractOutputStreamWriter abstractWriter && abstractWriter.getOutputStream() instanceof HttpOutput httpOutput)
+            httpOutput.complete(callback);
+        else
+            callback.succeeded();
```

Comment:
Is this enough? Should you not flush/close the non-Jetty writer?

Reply:
It is sufficient because we use our out `AbstractOutputStreamWriter` which we know does not cache characters (all done in `HttpOutput`).  I've updated comments to make it more apparent that this is the case.

Reply:
Not sure I understand. The `else` branch is the case when it's not our own Writer class, so feels that we should do something to forward the "completeness".

Also, turns out `ResponseWriter` is only created once, and with an `AbstractOutputStreamWriter`, so this check would be redundant.

If `ResponseWriter` is an internal mechanism, I would move everything internal and make sure `ResponseWriter` can only take `AbstractOutputStreamWriter` at the constructor.

Reply:
I think you are looking at old code. This method have refactored and we don't even pass in a callback anymore.